trollflow2/tests/test_launcher.py

```diff
@@ -773,6 +773,32 @@ def qsize(self):
         check_results(produced_files, start_time, -1)
     assert "Process killed with signal 1" in caplog.text
 
+    remote_filesystem = "s3://bucket-name"
+    caplog.clear()
+
+    produced_files = FakeQueue(0, 3)
+    with caplog.at_level(logging.DEBUG):
+        with mock.patch("trollflow2.launcher.S3FileSystem") as S3FileSystem:
+            S3FileSystem.return_value.stat.return_value = {'size': 0}
+            check_results(produced_files, start_time, 0, remote_filesystem=remote_filesystem)
+    assert "Empty file detected" in caplog.text
+    assert "files produced nominally" not in caplog.text
+
+    produced_files = FakeQueue(5, 8, skip=[6])
+    with caplog.at_level(logging.DEBUG):
+        with mock.patch("trollflow2.launcher.S3FileSystem") as S3FileSystem:
+            S3FileSystem.return_value.stat.side_effect = FileNotFoundError
+            check_results(produced_files, start_time, 0, remote_filesystem=remote_filesystem)
+    assert "Missing file" in caplog.text
+    assert "files produced nominally" not in caplog.text
+
+    produced_files = FakeQueue(10, 13)
+    with mock.patch("trollflow2.launcher.S3FileSystem") as S3FileSystem:
+        check_results(produced_files, start_time, 0, remote_filesystem=remote_filesystem)
+    assert mock.call('s3://bucket-name/file10') in S3FileSystem.return_value.stat.mock_calls
+    assert mock.call('s3://bucket-name/file11') in S3FileSystem.return_value.stat.mock_calls
+    assert mock.call('s3://bucket-name/file12') in S3FileSystem.return_value.stat.mock_calls
+
```

Comment:
These should be separate tests, the function here is already way too long :)

Reply:
I'd split the test, but I can't figure out where `tmp_path` is defined (guess it's a pytest fixture?) and how to pass it to `FakeQueue` without the now-used closure. I guess one option would be to use `tempfile` module, but I take it there's a reason to use the hidden `tmp_path` so don't want to go that way either.

Reply:
Split to separate tests in https://github.com/pytroll/trollflow2/pull/158/commits/ea049f1b94240f5199b9139a51b4ad6f1ea57ae0

The `FakeQueue` class didn't need a specific init interface, so the closure was never needed.